src/main/java/reposense/util/TimeUtil.java

```diff
@@ -26,6 +28,15 @@ public class TimeUtil {
     private static final String MESSAGE_SINCE_DATE_LATER_THAN_TODAY_DATE =
             "\"Since Date\" must not be later than today's date.";
 
+    private static final String MESSAGE_SINCE_DATE_EARLIER_THAN_EARLIEST_VALID_DATE =
+            "\"Since Date\" of %s must not be earlier than %s, resetting it to earliest valid date";
+    private static final String MESSAGE_UNTIL_DATE_LATER_THAN_LATEST_VALID_DATE =
+            "\"Until Date\" of %s must not be later than %s, resetting it to latest valid date";
+    private static final String EARLIEST_VALID_DATE = "1970-01-01T00:00:00";
+    private static final String LATEST_VALID_DATE = "2099-12-31T23:59:59";
```

Comment:
Sorry to clarify, do you mean to directly put the values of `EARLIEST_VALID_DATE`/`LATEST_VALID_DATE` into their respective message instead of separating them out? My main intention for separating them out was so that if the dates are changed, their respective messages do not need to be updated separately.

Reply:
I think keeping `EARLIEST_VALID_DATE` and `LATEST_VALID_DATE` as separate variables is good. What I meant is that you can directly do something like 
`MESSAGE_UNTIL_DATE_LATER_THAN_LATEST_VALID_DATE = String.format(..., LATEST_VALID_DATE)`

This is somewhat subjective but I think there is little reason the message template will be used differently.